### ISSUE TITLE
SideNav - support auto collapsing on mobile when Browser has been resized

### ISSUE THREAD

**Issue by Magallo** [GwtMaterialDesign/gwt-material#785]
Is there a way to auto collapse the sidenavbar when the horizontal size of the window is too small and the main content of the page is greyed (e.g. on mobile)?

I explain better: I am building an app with a skeleton similar to the showcase with a MaterialSideNavPush on the left of the screen and the rest of the screen is left for the main content which depends on what the user chooses in the side navigation panel. The left navigation panel is opened and showed by design. Using the smartphone (or resizing the browser to a very small page), the navigation panel is collapsed automatically and the user must interact with the menu icon to open it. When you open it, the main content is automatically greyed until you don't collapse again the side panel clicking again on the menu button. This is ok, but what I would like is something so that when the user chooses an item in the side panel, the main content changes and automatically the side panel collapses again showing the main content and auto hiding the side panel.

**Comment by kevzlou7979** [comment]
We could add this to the next release, For now the side nav we don't have that kind of dynamic collapsing when the browser resized. But we can listen into it, oh I think we can make use of the ViewPort feature to detect it. I will try to implement this on next release.

**Comment by kevzlou7979** [comment]
I've created a property called ``` autoHidesOnMobile="true" ``` wherein when enabled the sidenav will auto hide / collapse it durating browser resized.

**Comment by kevzlou7979** [comment]
## autoHideOnResize = true
![autohideonresize](https://user-images.githubusercontent.com/3138071/39037730-ef18bf38-44b3-11e8-97c4-4266dd0cde8a.gif)


## autoHideOnResize = false
![autohideonresize-yrd](https://user-images.githubusercontent.com/3138071/39037731-f02a1b60-44b3-11e8-81ee-e71541455fcc.gif)

**Comment by Magallo** [comment]
This is great. However I have question. Will this feature let the side panel automatically collapses showing the main content and auto hiding itself when the user chooses an item in the side panel which causes a navigation i.e. a change in the main content?

**Comment by Magallo** [comment]
Hi kevzlou7979. I tried the closeOnClick = true as you suggested but this is not a valid solution for the reasons I'm going to explain:
- closeOnClick = true collapse the side panel in all cases, in all situations. What I would like is to collapse it only if I have just expanded it in a mobile scenario where the main content is greyed because the horizontal size of the page is too small to show both side panel and main content. On the other hand, in a desktop environment where the window of the browser is, for example, maximized, and I have a lot of space and both the side panel and the main content are fully operational and not greyed, I don't want the side panel to be collapsed when I choose an item from it.
- closeOnClick = true causes the side panel to be collapsed even when I choose a MaterialLink inside a MaterialCollapsibleHeader which doesn't cause the navigation to another page but simply causes the expand/collapse the MaterialLink components inside its MaterialCollapsibleBody part.

In other words what I would like to get is (I try to explain again and try to be more clear):
- in a desktop scenario with a large horizontal size of the page, where both the side nav and the main content are shown, I would like the side nav to stay open.
- in a mobile scenario, i.e. when the side nav is open and the main content is greyed, when I choose a MaterialLink from the side nav which causes the main content to change, I would like the side nav to be collapsed and the main content to be enabled (not greyed). However, If I choose a MaterialLink which does not cause a change of the main content, i.e. a MaterialLink in a collapsible section, I would like the side nav panel to stay opened and the main content greyed.

What I explained above is just a behaviour a user would expect interacting with a app/web page which is mobile friendly.
